### PR TITLE
test(p7): close coverage gaps in storyboard, design HTML, image mapping, prompt assembly

### PR DESCRIPTION
**Pull request by cyberprophet**
## Summary

This PR closes meaningful coverage gaps identified by a systematic audit of the P7 Agency library public surface vs. existing test files. The following areas had zero direct unit tests despite containing significant logic ported from P3 shop-plugin (ADR-008) or written in the current session:

| Gap | Risk |
|-----|------|
| `ValidateStoryboard` (8 gates) | Regressions in validation let invalid storyboards reach P5/image generation pipeline |
| `AutoCorrectStoryboard` | A broken correction corrupts final-attempt storyboards silently |
| `BuildUserMessage` / `BuildBlueprintUserMessage` | Prompt assembly errors lose model context (wrong sections, missing data, unescaped injection) |
| `ValidateDesignHtml` (gates 1-3, 6) | Invalid HTML bypasses section-count and content checks |
| `SanitizeHtml` (gates 4-5) | XSS vectors (`<script>`, `on*` handlers) reach the frontend |
| `GptService.MapSize` | Wrong canvas causes every image to render at incorrect aspect ratio |
| `PromptSanitizer.EscapeIdentifierForPrompt` | HTML-injection in document ids could break product-info extraction |
| `StripMarkdownFences` | Broken fence stripping returns unparseable JSON for Visual DNA / research results |

## Tests added (103 new, all green)

- **`StoryboardValidationTests.cs`** (36 tests) — `ValidateStoryboard` all 8 gates, `AutoCorrectStoryboard` 5 scenarios, `BuildUserMessage` 8 prompt-assembly tests.
- **`DesignHtmlValidationTests.cs`** (26 tests) — `ValidateDesignHtml` gates 1-3+6, `SanitizeHtml` gates 4-5, edge cases and integration.
- **`ImageMapSizeTests.cs`** + **`EscapeIdentifierForPromptTests`** (21 tests) — `MapSize` all 3 branches + fallback invariants, `EscapeIdentifierForPrompt` HTML special chars + injection resistance.
- **`BlueprintUserMessageTests.cs`** + **`StripMarkdownFencesTests`** (20 tests) — `BuildBlueprintUserMessage` section structure and ordering, `StripMarkdownFences` fence variants.

## Test run

```
Passed! - Failed: 0, Passed: 334 (103 new), Skipped: 0, Total: 334
```

6 pre-existing `EmbeddedResourceTests` failures on `main` (missing embedded Prompts resources) are unchanged by this PR.

## Constraints respected

- NuGet version not bumped (stays at 0.9.1).
- No publish to nuget.org.
- No production code modified.
- All tests exercise actual code paths via `internal` access or reflection (consistent with existing `BlueprintValidationTests`/`WebToolsHtmlExtractionTests` conventions).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: agency.tests/StoryboardValidationTests.cs [hunk 1]
```diff
@@ -0,0 +1,726 @@
...
+                SectionWithImage("FAQ", "Answer questions", "faq"),
+                SectionWithImage("Specs", "Show specs", "spec-table"),
+            ],
+            CtaText: "Order Now");
+
+        var error = Validate(storyboard, autoCorrect: true);
+
+        // autoCorrect=true suppresses gate-1 image-block error
+        if (error is not null)
+            Assert.DoesNotContain("missing type: \"image\" blocks", error);
```
Comment (cyberchacha) [inline agency.tests/StoryboardValidationTests.cs:136]: **<sub><sub>P3 Badge</sub></sub>  Weak assertion pattern for negative gate tests**

The `if (error is not null) Assert.DoesNotContain(...)` pattern silently passes when `Validate` returns `null` (no error at all), which is fine. However, it also silently passes when a *different* gate fires an error that doesn't contain the checked substring. If a future refactor causes a different gate to fire for this input, the test would still pass green without catching the regression.

Consider adding `Assert.Null(error)` for happy-path tests where no error is expected at all, and reserving the `DoesNotContain` pattern only for cases where a *different* gate error is acceptable (e.g., line 449 for digital products that may fail other gates).

This is a minor robustness concern — the tests are correct as-is for current behavior.

Useful? React with 👍 / 👎.

File: agency.tests/StoryboardValidationTests.cs [hunk 2]
```diff
@@ -0,0 +1,726 @@
...
+        Assert.NotNull(error);
+        Assert.Contains("too short", error);
+        Assert.Contains("min 100", error);
+    }
+
+    [Fact]
+    public void Validate_ImagePromptAtMinimumLength_NoLengthError()
+    {
+        // Exactly 100 chars
+        var prompt = new string('A', 90) + " on white background setup.";
```
Comment (cyberchacha) [inline agency.tests/StoryboardValidationTests.cs:211]: **<sub><sub>P3 Badge</sub></sub>  Boundary test prompt length is fragile**

The "exactly 100 chars" boundary test constructs the string as `new string('A', 90) + " on white background setup."` and then asserts `prompt.Trim().Length >= 100`. The appended suffix is 27 chars, giving 117 total — well above 100, not exactly at the boundary. If the intent is to test the exact boundary (100 chars), consider constructing a string of exactly 100 characters to ensure the boundary condition is properly exercised.

Useful? React with 👍 / 👎.

File: agency.tests/BlueprintUserMessageTests.cs [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+        // The storyboard content is in the first "## Storyboard" section.
+        // Collect text up to the next "## " section header.
+        var storyboardSection = sections.FirstOrDefault(s => s.StartsWith("Storyboard")) ?? string.Empty;
+
+        // Strip the trailing legitimate </user_input> and confirm no raw close tag remains
+        var lastClose = storyboardSection.LastIndexOf("</user_input>", StringComparison.Ordinal);
+        if (lastClose >= 0)
+        {
+            var before = storyboardSection[..lastClose];
+            Assert.DoesNotContain("</user_input>", before);
```
Comment (cyberchacha) [inline agency.tests/BlueprintUserMessageTests.cs:199]: **<sub><sub>P3 Badge</sub></sub>  Prompt injection test could be more explicit**

The injection test strips the last legitimate `</user_input>` and checks the remainder doesn't contain a raw close tag. This works, but it implicitly depends on the sanitizer's escaping strategy. A more explicit assertion — e.g., verifying the injected `</user_input>` is HTML-encoded as `&lt;/user_input&gt;` — would make the test intent clearer and more resilient to changes in the escaping approach.

Useful? React with 👍 / 👎.

**Review by cyberchacha — APPROVED** [review]
## Review Summary
- Outcome: approve
- Summary: Well-structured test-only PR adding 103 tests that close meaningful coverage gaps across storyboard validation, design HTML sanitization, image mapping, and prompt assembly. No production code modified. Test patterns are consistent with existing conventions (reflection-based access to internal/private methods). A couple of minor observations noted inline but nothing blocking.
